### DISCUSSION TITLE
Keep connection alive while app is in background

### DISCUSSION THREAD

**Discussion by Symbai** [dotnet/maui#12596] [Q&A]
How do I keep a TCP connection alive when the user moves the app in background (both, Android and iOS) ? There seems to be no example for MAUI, all I found was some really old stuff for Xamarin which I cannot use on Maui.

**Comment by Ghevi** [comment]
I would like to know too, i'm new to mobile development so take this with a grain of salt.
From what i've read online and tried, it seems like apps can be suspended/killed for a number of reasons. iOS is very restrictive: https://developer.apple.com/forums/thread/685525.
Android allow you more freedom but then custom rom from manufacturers like Xiaomi, Samsung ecc also restrict apps in their own ways and whitelist the most popular ones, like whatsapp.
I've manage to run some code every 15 minutes on Android but in the end I've opted for push notifications because on iOS there is no way to my knowledge to archieve this.
